### PR TITLE
[#2542] Use tenant trace sampling mode in Command Router

### LINKED ISSUE #2542: Enable forced sampling of traces for a specific tenant (Command Router)
The new config options introduced in #896 to enable forced sampling of traces (i.e. recording of traces in the tracing backend) for a specific tenant should also be taken into account in the Command Router.

### PR DESCRIPTION
**Pull request by calohmn**
This fixes #2542.

### CODE REVIEW

File: services/command-router-base/src/main/java/org/eclipse/hono/commandrouter/impl/amqp/ProtonBasedCommandConsumerFactoryImpl.java [hunk 1]
```diff
@@ -82,9 +83,10 @@ public ProtonBasedCommandConsumerFactoryImpl(
     }
 
     @Override
-    public void initialize(final CommandTargetMapper commandTargetMapper) {
+    public void initialize(final TenantClient tenantClient, final CommandTargetMapper commandTargetMapper) {
```
Comment (sophokles73) [inline services/command-router-base/src/main/java/org/eclipse/hono/commandrouter/impl/amqp/ProtonBasedCommandConsumerFactoryImpl.java:86]: can't we simply add *tenantClient* and *commandTargetMapper* to the constructor and get rid of this method? We already have life cycle methods and having an additional *initialize* before we can actually start the component seems really awkward to me ...

File: services/command-router-base/src/main/java/org/eclipse/hono/commandrouter/impl/kafka/KafkaBasedCommandConsumerFactoryImpl.java [hunk 2]
```diff
@@ -112,10 +113,11 @@ public KafkaBasedCommandConsumerFactoryImpl(
     }
 
     @Override
-    public void initialize(final CommandTargetMapper commandTargetMapper) {
+    public void initialize(final TenantClient tenantClient, final CommandTargetMapper commandTargetMapper) {
```
Comment (sophokles73) [inline services/command-router-base/src/main/java/org/eclipse/hono/commandrouter/impl/kafka/KafkaBasedCommandConsumerFactoryImpl.java:116]: same here, IMHO we can get rid of this method, can't we?

File: services/command-router-base/src/main/java/org/eclipse/hono/commandrouter/impl/amqp/ProtonBasedCommandConsumerFactoryImpl.java (hunk 1 above)
Comment (calohmn) [reply services/command-router-base/src/main/java/org/eclipse/hono/commandrouter/impl/amqp/ProtonBasedCommandConsumerFactoryImpl.java]: I've refactored the corresponding classes now to remove the `initialize` methods, also removing `@Autowired` annotations in `CommandRouterServiceImpl` (in the command router base module).

File: services/command-router-base/src/main/java/org/eclipse/hono/commandrouter/impl/CommandRouterServiceImpl.java [hunk 3]
```diff
@@ -53,82 +48,38 @@
...
-     * If not set explicitly, the {@code NoopTracer} from OpenTracing will
-     * be used.
+     * Creates a new CommandRouterServiceImpl.
      *
-     * @param opentracingTracer The tracer.
+     * @param config The command router service configuration.
+     * @param registrationClient The device registration client.
+     * @param tenantClient The tenant client.
+     * @param deviceConnectionInfo The client for accessing device connection data.
+     * @param commandConsumerFactory The factory to use for creating clients to receive commands.
```
Comment (sophokles73) [inline services/command-router-base/src/main/java/org/eclipse/hono/commandrouter/impl/CommandRouterServiceImpl.java:69]: IMHO we should document the NullPointerException if params are `null`

**Review by sophokles73 — APPROVED** [review]
LGTM

**Pull request merged**